zio-schema/shared/src/main/scala/zio/schema/SchemaEquality.scala

```diff
@@ -0,0 +1,76 @@
+package zio.schema
+
+import zio.prelude._
+
+trait SchemaEquality {
+
+  val strictEquality: Equal[Schema[_]] =
+    new Equal[Schema[_]] {
+      override protected def checkEqual(l: Schema[_], r: Schema[_]): Boolean =
+        recursiveEqual(l, r, Set.empty)
+
+      private def recursiveEqual(l: Schema[_], r: Schema[_], visitedLazy: Set[Schema.Lazy[_]]): Boolean = {
+        implicit lazy val selfEqual: Equal[Schema[_]] = Equal.make(recursiveEqual(_, _, visitedLazy))
+        implicit lazy val fieldEqual: Equal[Schema.Field[_]] =
+          (l: Schema.Field[_], r: Schema.Field[_]) => {
+            l.label === r.label &&
+              l.schema === r.schema &&
+              l.annotations == r.annotations
+          }
+
+        (l, r) match {
+          case (lEnum: Schema.Enum[_], rEnum: Schema.Enum[_]) =>
+            l.annotations == r.annotations && lEnum.structure === rEnum.structure
+          case (lRecord: Schema.Record[_], rRecord: Schema.Record[_]) =>
+            l.annotations == r.annotations && lRecord.structure === rRecord.structure
+          case (lMap: Schema.MapSchema[_, _], rMap: Schema.MapSchema[_, _]) =>
+            lMap.annotations == rMap.annotations && lMap.ks === rMap.ks && lMap.vs === rMap.vs
+          case (lSet: Schema.SetSchema[_], rSet: Schema.SetSchema[_]) =>
+            lSet.annotations == rSet.annotations && lSet.as === rSet.as
+          case (lSeq: Schema.Sequence[_, _], rSeq: Schema.Sequence[_, _]) =>
+            (lSeq.fromChunk eq rSeq.fromChunk) &&
+              (lSeq.toChunk eq rSeq.toChunk) &&
```

Comment:
Wouldn't this effectively just turn schema equality into reference equality (at least for `Sequence`)?

Reply:
Those are conversion functions like in `Transform`. Probably that `identity` trick should be applied for sequence too, or fixed into `Chunk` and other collection types encoded as a combination of `Sequence` and `Transform`

Reply:
I first wanted to just get rid of the conversion and use `Transform` everywhere but it made hard to create the traversal optics so I just did the same extra identity field as with transform.